geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSCardIntegrationTest.java

```diff
@@ -0,0 +1,154 @@
+    jedis.sadd(key, values);
+    assertThat(jedis.scard(key)).isEqualTo(values.length);
+  }
+
+  @Test
+  public void scardNonExistentSet_returnsZero() {
+    assertThat(jedis.scard("{user1}nonExistentSet")).isEqualTo(0);
+  }
+
+  @Test
+  public void scardAfterSadd_returnsSize() {
+    String[] values = new String[] {"Jeff", "Natalie", "Michelle", "Joe", "Kelley"};
+
+    assertThat(jedis.scard(key)).isEqualTo(0);
+    jedis.sadd(key, values);
+
+    jedis.sadd(key, values);
+    assertThat(jedis.scard(key)).isEqualTo(values.length);
+  }
+
+  @Test
+  public void scardConcurrentSAddSCard_sameKeyPerClient()
+      throws InterruptedException, ExecutionException {
+    int elements = 1000;
+
+    String key = generator.generate('x');
+    String[] members1 = generateStrings(elements, 'y');
+    String[] members2 = generateStrings(elements, 'z');
+
+    ExecutorService pool = Executors.newFixedThreadPool(2);
+    Callable<Integer> callable1 = () -> doABunchOfSAdds(key, members1, jedis);
+    Callable<Integer> callable2 = () -> doABunchOfSAdds(key, members2, jedis);
+    Future<Integer> future1 = pool.submit(callable1);
+    Future<Integer> future2 = pool.submit(callable2);
+
+    Assertions.assertThat(future1.get()).isEqualTo(members1.length);
+    Assertions.assertThat(future2.get()).isEqualTo(members2.length);
+
+    Assertions.assertThat(jedis.scard(key)).isEqualTo(members1.length + members2.length);
+
+    pool.shutdown();
+  }
+
+  @Test
+  public void scardConcurrentSAddSCard_differentKeyPerClient()
+      throws InterruptedException, ExecutionException {
+    int elements = 1000;
+    String key1 = generator.generate('x');
+    String key2 = generator.generate('y');
+
+    String[] strings = generateStrings(elements, 'y');
+
+    ExecutorService pool = Executors.newFixedThreadPool(2);
+    Callable<Integer> callable1 = () -> doABunchOfSAdds(key1, strings, jedis);
+    Callable<Integer> callable2 = () -> doABunchOfSAdds(key2, strings, jedis);
+    Future<Integer> future1 = pool.submit(callable1);
+    Future<Integer> future2 = pool.submit(callable2);
+
+    Assertions.assertThat(future1.get()).isEqualTo(strings.length);
+    Assertions.assertThat(future2.get()).isEqualTo(strings.length);
+
+    Assertions.assertThat(jedis.scard(key1)).isEqualTo(strings.length);
+    Assertions.assertThat(jedis.scard(key2)).isEqualTo(strings.length);
+
+    pool.shutdown();
+  }
+
+  private int doABunchOfSAdds(String key, String[] strings, JedisCluster jedis) {
+    int successes = 0;
+
+    for (int i = 0; i < strings.length; i++) {
+      Long reply = jedis.sadd(key, strings[i]);
+      if (reply == 1L) {
+        successes++;
+        Thread.yield();
+      }
+    }
+    return successes;
+  }
+
+  private String[] generateStrings(int elements, char uniqueElement) {
+    Set<String> strings = new HashSet<>();
+    for (int i = 0; i < elements; i++) {
+      String elem = generator.generate(uniqueElement);
```

Comment:
A simpler way of generating unique members is to just use some pre-defined base and add the value of i to it:
```
  private String[] generateStrings(int elements) {
    String[] strings = new String[elements];
    for (int i = 0; i < elements; i++) {
      strings[i] = "member" + i;
    }
    return strings;
  }
```

Reply:
Removed this code since I am modifying the concurrent testing.